code/optimize/src/test/java/com/adobe/marketing/mobile/optimize/XDMUtilsTest.kt

```diff
@@ -0,0 +1,140 @@
+                        }
+                    } ?: false
+                }
+
+            expectedStructure is List<*> && actual is List<*> ->
+                expectedStructure.firstOrNull()?.let { expectedValue ->
+                    actual.firstOrNull()?.let { actualValue ->
+                        validateStructure(expectedValue, actualValue)
+                    }
+                } ?: false
+
+            else -> expectedStructure::class == actual::class
+        }
+    }
+
+    @Before
+    fun setUp() {
+        val multipleItemsList: List<Map<String, Any>> =
+            loadJsonFromFile("json/MULTIPLE_PROPOSITION_VALID_TARGET.json") ?: emptyList()
+        val singlePropositionMap: Map<String, Any> =
+            loadJsonFromFile("json/PROPOSITION_VALID_TARGET.json") ?: emptyMap()
+        singleProposition = OptimizeProposition.fromEventData(singlePropositionMap)
+        multiplePropositions = multipleItemsList.map { OptimizeProposition.fromEventData(it) }
+    }
+
+    @After
+    fun tearDown() {
+        unmockkAll()
+    }
+
+    @Test
+    fun `generateInteractionXdm should return correct structure for SingleProposition`() {
+        val experienceEventType = "decisioning.propositionDisplay"
+        val result = XDMUtils.generateInteractionXdm(experienceEventType, listOf(singleProposition))
+
+        val expectedStructure = mapOf(
+            "_experience" to mapOf(
+                "decisioning" to mapOf(
+                    "propositions" to listOf(
+                        mapOf(
+                            "id" to "",
+                            "scope" to "",
+                            "scopeDetails" to mapOf<String, Any>(),
+                            "items" to listOf(
+                                mapOf("id" to "")
+                            )
+                        )
+                    )
+                )
+            ),
+            "eventType" to ""
+        )
+
+        Assert.assertTrue(validateStructure(expectedStructure, result))
+    }
+
+    @Test
+    fun `generateInteractionXdm should return correct structure for MultiplePropositions`() {
+        val experienceEventType = "decisioning.propositionDisplay"
+        val result = XDMUtils.generateInteractionXdm(experienceEventType, multiplePropositions)
+
+        val expectedStructure = mapOf(
+            "_experience" to mapOf(
+                "decisioning" to mapOf(
+                    "propositions" to listOf(
+                        mapOf(
+                            "id" to "",
+                            "scope" to "",
+                            "scopeDetails" to mapOf<String, Any>()
+                        )
+                    )
+                )
+            ),
+            "eventType" to ""
+        )
+
+        Assert.assertTrue(validateStructure(expectedStructure, result))
+    }
+
+    @Test
+    fun `generateInteractionXdm should not include items array for MultiplePropositions`() {
```

Comment:
nit: Is this test title supposed to be `generateInteractionXdm should include items array for MultiplePropositions`?